bindings/wasm/examples/src/1_advanced/7_status_list_2021.ts

```diff
@@ -0,0 +1,157 @@
+    const statusList = new StatusList2021();
+
+    // Create a status list credential so that the status list can be stored anywhere.
+    // In this example the credential will fictitiously be made available at `http://example.com/credential/status`
+    // (actually it will stay in memory).
+    const statusListCredential = new StatusList2021CredentialBuilder(statusList)
+        .purpose(StatusPurpose.Revocation)
+        .subjectId("http://example.com/credential/status")
+        .issuer(issuerDocument.id().toString())
+        .build();
+
+    // Create a credential subject indicating the degree earned by Alice, linked to their DID.
+    const subject = {
+        id: aliceDocument.id(),
+        name: "Alice",
+        degreeName: "Bachelor of Science and Arts",
+        degreeType: "BachelorDegree",
+        GPA: "4.0",
+    };
+
+    // Create an unsigned `UniversityDegree` credential for Alice.
+    // The issuer also chooses a unique `StatusList2021` index to be able to revoke it later.
+    const CREDENTIAL_INDEX = 5;
+    const status = new StatusList2021Entry(statusListCredential.id(), statusListCredential.purpose(), CREDENTIAL_INDEX)
+        .toStatus();
+    const credential = new Credential({
+        id: "https://example.edu/credentials/3732",
+        type: "UniversityDegreeCredential",
+        credentialStatus: status,
+        issuer: issuerDocument.id(),
+        credentialSubject: subject,
+    });
+
+    // Create signed JWT credential.
+    const credentialJwt = await issuerDocument.createCredentialJwt(
+        issuerStorage,
+        issuerFragment,
+        credential,
+        new JwsSignatureOptions(),
+    );
+    console.log(`Credential JWT > ${credentialJwt.toString()}`);
+
+    // Validate the credential using the issuer's DID Document.
+    const validationOptions = new JwtCredentialValidationOptions({ status: StatusCheck.SkipUnsupported });
+    let jwtCredentialValidator = new JwtCredentialValidator(new Ed25519JwsVerifier());
+    jwtCredentialValidator.validate(
+        credentialJwt,
+        issuerDocument,
+        validationOptions,
+        FailFast.FirstError,
+    );
+
+    // ===========================================================================
+    // Revocation of the Verifiable Credential.
+    // ===========================================================================
+
+    // Update the status list credential.
+    // This revokes the credential's unique index.
+    statusListCredential.setCredentialStatus(credential, CREDENTIAL_INDEX, true);
+
+    // Credential verification now fails.
+    try {
+        jwtCredentialValidator.validate(
+            credentialJwt,
+            issuerDocument,
+            validationOptions,
+            FailFast.FirstError,
+        );
+        JwtCredentialValidator.checkStatusWithStatusList2021(credential, statusListCredential, StatusCheck.Strict);
+        console.log("Revocation Failed!");
```

Comment:
This is a bid ambiguous

Reply:
Copied this from `vc_revocation` example. How can I improve it? Shall I add a comment specifying that since the credential has been revoked `checkStatusWithStatusList2021` should throw an exception?